code/modules/SCP/scp017.dm

```diff
@@ -0,0 +1,637 @@
+		StopEnveloping(FALSE)
+		LightSourceDebuff() // Apply debuff for being hit by strong light
+		SafeVisibleMessage(span_notice("[src] recoils from the intense light, releasing [enveloping_target]!"))
+		return TRUE // Consume the projectile
+	if(Proj.damage_type == BRUTE)
+		SafeVisibleMessage(span_warning("The [Proj] seems to pass right through it!"))
+		return TRUE
+	return ..()
+
+/mob/living/simple_animal/hostile/scp017/proc/LightSourceDebuff()
+	if(!HAS_TRAIT(src, TRAIT_IMMOBILIZED))
+		ADD_TRAIT(src, TRAIT_IMMOBILIZED, "SCP017_LIGHT_DEBUFF")
+		shadow_threshold -= 0.15 // Decrease threshold, making it vulnerable in slightly brighter areas
+		SafeVisibleMessage(span_warning("[src] shimmers erratically as it absorbs the light!"))
+		addtimer(CALLBACK(src, PROC_REF(RemoveLightSourceDebuff)), light_source_debuff_duration)
+
+/mob/living/simple_animal/hostile/scp017/proc/RemoveLightSourceDebuff()
+	if(HAS_TRAIT(src, TRAIT_IMMOBILIZED))
+		REMOVE_TRAIT(src, TRAIT_IMMOBILIZED, "SCP017_LIGHT_DEBUFF")
+		shadow_threshold = initial(shadow_threshold) // Reset threshold
+		SafeVisibleMessage(span_notice("[src] stabilizes as the light's influence fades."))
+
+/mob/living/simple_animal/hostile/scp017/proc/SafeVisibleMessage(message_text, message_type)
+	if(world.time >= last_message_time + message_cooldown)
+		visible_message(message_text)
+		last_message_time = world.time
+
+//Attack
+
+/mob/living/simple_animal/hostile/scp017/UnarmedAttack(atom/A)
+	if(!CanAttack(A))
+		return FALSE
+
+	var/turf/Tturf = get_turf(A)
+	if(!Tturf)
+		return FALSE
+
+	do_smoke(amount = 1, location = Tturf)
+
+	var/death_message = pick("[A] disappears into [src]!", "[A] is enveloped by [src]!", "[A] is absorbed by [src]!")
+	SafeVisibleMessage(span_danger(death_message))
+
+	if(ismob(A))
+		var/mob/living/target_mob = A
+		StartEnveloping(target_mob)
+	else
+		qdel(A)
+
+//General Interactions
+/mob/living/simple_animal/hostile/scp017/proc/StartEnveloping(mob/living/target_mob)
+	if(enveloping_target) // Already enveloping someone
+		return FALSE
+
+	enveloping_target = target_mob
+	SafeVisibleMessage(span_danger("[enveloping_target] is being enveloped by [src]!"))
+
+	// Apply initial vision reduction and blurriness
+	if(enveloping_target.client)
+		enveloping_target.client.eye = src // Force eye to SCP-017 for effect
+		enveloping_target.see_in_dark = max(0, enveloping_target.see_in_dark - 5) // Initial vision reduction
+		enveloping_target.adjust_blurriness(0.2) // Initial blurriness
+
+	enveloping_progress_timer_id = addtimer(CALLBACK(src, PROC_REF(UpdateEnveloping)), 10, enveloping_duration / 10) // Call every 1 second for duration
+
+	return TRUE
+
+/mob/living/simple_animal/hostile/scp017/proc/UpdateEnveloping()
+	if(!enveloping_target || QDELETED(enveloping_target))
+		StopEnveloping()
+		return
+
+	if(enveloping_target.client)
+		enveloping_target.see_in_dark = max(0, enveloping_target.see_in_dark - 5) // Progressively reduce vision
```

Comment:
`update_sight()` will immediately undo this, you will want to make this a status effect and hook it into said proc.